src/mbgl/renderer/layers/render_circle_layer.cpp

```diff
@@ -187,4 +187,31 @@ bool RenderCircleLayer::queryIntersectsFeature(
     return false;
 }
 
+RenderLayer::RenderTiles RenderCircleLayer::filterRenderTiles(RenderTiles tiles) const {
+    RenderTiles filteredTiles = RenderLayer::filterRenderTiles(tiles);
+
+    // Tiles are sorted by zoom level, if first and last tiles are from different zoom
+    // levels, check whether tile vector contains parents covering children area.
+    // This might happen when map is panned (or zoomed) and missing tile is not yet loaded,
+    // thus, cached parent tile is used to cover missing area, therefore, loaded child tiles
+    // will be drawn together with the parent.
+
+    if (filteredTiles.size() > 1 &&
+            filteredTiles.front().get().id.canonical.z != filteredTiles.back().get().id.canonical.z) {
+
+        // Filter out children tiles if there is a parent covering same area.
+        for (auto& parentTile : filteredTiles) {
+            filteredTiles.erase(
```

Comment:
`RenderTiles` is a `std::vector` and I don't think it is safe to call `erase` while iterating.

Reply:
:+1:  + rewrote that piece of code to be a bit more efficient.